Understanding WebAssembly better by learning WebAssembly-Text.md

```diff
@@ -46,29 +48,30 @@ Well, the point is that WASM wasn't created to print strings and interact with
 APIs, its purpose is to help JavaScript handle heavy computations by providing
 an interface to fast, low level instructions.
 ### But what does the code do?
+ - `i32` is one of the four primary types of WebAssembly, it's a 32-bit integer type.
  - `func` declares a function
- - `$hello_world` is a compile time name/label we give the function (we'll see that later)
- - `(param $0 i32)` and `(param $1 i32)` tell that this function accepts two parameters, the first one labeled $0 (notice the `$`) with a type `i32`,
- 	the second one labeled $1 with a type `i32`.
+ - `$hello_world` is a compile time name/label we give the function (we'll see more about that later)
+ - `(param $lhs i32)` and `(param $rhs i32)` tell that this function accepts two parameters, the first one labeled $lhs for left-hand side (notice the `$`) with a type of `i32`,
+ 	the second one labeled $rhs for right-hand side with a type of `i32`.
  - `(result i32)` says that the function return type is an `i32`.
- - `local.get $0` pushes the value of the parameter labeled as $0 into the stack.
- - `local.get $1` does the same as above, but instead it pushes the value of $1.
+ - `local.get $lhs` pushes the value of the parameter labeled as $lhs into the stack.
+ - `local.get $rhs` does the same as above, but instead it pushes the value of $rhs.
  - `i32.add` pops two values of type `i32` from the stack and pushes the result of their addition back into the stack.
  - `(export "helloWorld" (func $hello_world))` exports the function labeled as `$hello_world` to the host with the name "helloWorld".
 
 ### Where is the `return` statement
 WebAssembly has a `return` instruction, but it is mostly used when you need
 to immediatly return and stop executing the function any further. Otherwise,
-there will be an implicit return, where the last value is poped from the stack
+there will be an implicit return, where the last value is popped from the stack
 and returned.
 
 ### What does a "label" actually mean?
-All the function calls, parameter and local acces are done by index. Labels
+All the function calls, parameter and local access are done by index. Labels
 are just compile-time annotations to make code easier to read, write and
 understand.
 
 ### Are `local.get` and `i32.add` the only instructions?
-Of course not. The WebAssembly instruction set is really big. Just the MVP
+Of course not. The WebAssembly instruction set is *huge*. Just the MVP (minimal viable product)
 has around 120 instructions. Most of them start with the type, which can only
 be `i32`, `i64`, `f32`, `f64`, followed by a dot and the name of the
 operation. There are also instructions for other purpouses, like control flow,
```

Comment:
Mentioned control flow here, but later on you simplified a subtitle to "Decision making"